src/base/document.js

```diff
@@ -0,0 +1,109 @@
+import { DocumentsManager } from '../api2'
+
+/**
+ * Wrapper to simplify work with documents
+ * @class
+ */
+export class Document {
+  /**
+   * @param {object} opts
+   * @param {File} [opts.file] - file representing a document
+   * @param {string} [opts.name] - file name
+   * @param {string} [opts.mimeType] - file MIME type
+   * @param {string} [opts.key] - storage service file key
+   * @param {string} [type] - storage service document type
+   */
+  constructor (opts = {}, type = '') {
+    this._file = opts.file || null
+    this._name = opts.name || ''
+    this._mimeType = opts.mimeType || ''
+    this._key = opts.key || ''
+    this._type = type || ''
+    this._privateUrl = ''
+  }
+
+  get file () { return this._file }
+  get name () { return this._name }
+  get mimeType () { return this._mimeType }
+  get key () { return this._key }
+  get type () { return this._type }
+  get privateUrl () { return this._privateUrl }
+
+  get publicUrl () {
+    return this.documentsManager.getDocumentUrlByKey(this._key)
+  }
+
+  get isEmpty () {
+    return !this._file && !this.isUploaded
+  }
+
+  get isUploaded () {
+    return Boolean(this._key)
+  }
+
+  setFile (val) {
+    this._file = val
+    return this
+  }
+
+  setName (val) {
+    this._name = val
+    return this
+  }
+
+  setMimeType (val) {
+    this._mimeType = val
+    return this
+  }
+
+  setKey (val) {
+    this._key = val
+    return this
+  }
+
+  setType (val) {
+    this._type = val
+    return this
+  }
+
+  toJSON () {
+    return {
+      mime_type: this._mimeType || '',
+      name: this._name || '',
+      key: this._key || ''
+    }
+  }
+
+  static toEmptyJSON () {
+    return {
+      mime_type: '',
+      name: '',
+      key: ''
+    }
+  }
+
+  async uploadSelf () {
+    if (this.isUploaded) return this
```

Comment:
Oh, it's also did policies conversion before, now it seems to be missing

Reply:
we can pass them to `documentsManager` when initializing it, and do the conversion inside it

Reply:
need to discuss